neqo-crypto/src/agent.rs

```diff
@@ -40,16 +43,67 @@ use crate::{
     },
     ech,
     err::{is_blocked, secstatus_to_res, Error, PRErrorCode, Res},
+    experimental_api,
     ext::{ExtensionHandler, ExtensionTracker},
-    null_safe_slice,
     p11::{self, PrivateKey, PublicKey},
     prio,
     replay::AntiReplay,
     secrets::SecretHolder,
-    ssl::{self, PRBool},
+    ssl::{self, PRBool, SSLCertificateCompressionAlgorithm},
     time::{Time, TimeHolder},
 };
 
+experimental_api!(SSL_SetCertificateCompressionAlgorithm(
+    fd: *mut ssl::PRFileDesc,
+    t: SSLCertificateCompressionAlgorithm,
+));
+
+/// The trait is used to represent a certificate compression data structure
+/// Used in order to enable Certificate Compression extension during TLS connection
+pub trait CertificateCompressor {
+    /// Certificate Compression Algorithm identifier
+    /// zlib(1), brotli(2), zstd(3),
+    fn id(&self) -> u16;
+    /// Name of the certificate compression algorithm
+    fn name(&self) -> &'static str;
+
+    /// The function that's used for encoding the certificate
+    /// Arguments:
+    ///
+    /// * `input`: Pointer to SECItem containing the bytes to encode
+    /// * `output`: Pointer to SECItem to store the result
+    ///
+    /// Note:  Certificate Compression encoding function is responsible
+    /// for allocating the output buffer itself.
+
+    fn encode(
```

Comment:
This is not an encode function, but a getter for an encoder:

```suggestion
    fn encoder(
```